orca/src/llm/openai.rs

```diff
@@ -90,13 +104,29 @@ impl Display for Response {
     }
 }
 
+impl Display for JsonModeResponse {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut s = String::new();
+        for choice in &self.choices {
+            s.push_str(&choice.message.content);
+        }
+        write!(f, "{}", s)
+    }
+}
+
 #[derive(Serialize, Deserialize, Debug, Default, Clone)]
 pub struct Usage {
     prompt_tokens: i32,
     completion_tokens: Option<i32>,
     total_tokens: i32,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ResponseFormat {
+    #[serde(rename = "type")]
+    pub type_: String,
```

Comment:
type has a finite combination of states. Sounds 
```suggestion
    pub type_: ResponseType,
```
 I don't want users to remember what is the exact string they type. Is it "json_object", "JSON", ".json"?

Reply:
Great point, I changed it to an enum and set the default to TEXT. So if they just want another type they can just pass through the ResponseFormat they would like :)